qiskit/primitives/statevector_sampler.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Iterable
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit import ClassicalRegister, QiskitError, QuantumCircuit
+from qiskit.circuit import ControlFlowOp
+from qiskit.quantum_info import Statevector
+
+from .base import BaseSamplerV2
+from .base.validation import _has_measure
+from .containers import (
+    BitArray,
+    PrimitiveResult,
+    PubResult,
+    SamplerPub,
+    SamplerPubLike,
+    make_data_bin,
+)
+from .containers.bit_array import _min_num_bytes
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class _MeasureInfo:
+    creg_name: str
+    num_bits: int
+    num_bytes: int
+    qreg_indices: list[int]
+
+
+class Sampler(BaseSamplerV2):
+    """
+    Simple implementation of :class:`BaseSamplerV2` with Statevector.
+    """
+
+    _DEFAULT_SHOTS: int = 512
+
+    def __init__(self, *, seed: np.random.Generator | int | None = None):
+        """
+        Args:
+            seed: The seed for random number generator.
+        """
+        self._seed = seed
+        if isinstance(self._seed, np.random.Generator):
+            self._rng = self._seed
+        else:
+            self._rng = np.random.default_rng(self._seed)
+
+    @property
+    def seed(self) -> np.random.Generator | int | None:
+        """Return the seed for random number generator.
+
+        Returns:
+            np.random.Generator | int | None: The seed for random number generator.
+        """
+        return self._seed
+
+    def run(
+        self, pubs: Iterable[SamplerPubLike], *, shots: int | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        job: PrimitiveJob[PubResult] = PrimitiveJob(self._run, pubs, shots)
+        job._submit()
+        return job
+
+    def _run(
+        self, pubs: Iterable[SamplerPubLike], shots: int | None = None
+    ) -> PrimitiveResult[PrimitiveResult[PubResult]]:
+        if shots is None:
+            shots = self._DEFAULT_SHOTS
+        coerced_pubs = [SamplerPub.coerce(pub, shots) for pub in pubs]
+        for pub in coerced_pubs:
+            pub.validate()
+
+        results = []
+        for pub in coerced_pubs:
+            circuit, qargs, meas_info = _preprocess_circuit(pub.circuit)
+            bound_circuits = pub.parameter_values.bind_all(circuit)
+            arrays = {
+                item.creg_name: np.zeros(
+                    bound_circuits.shape + (pub.shots, item.num_bytes), dtype=np.uint8
+                )
+                for item in meas_info
+            }
+            for index in np.ndindex(*bound_circuits.shape):
+                bound_circuit = bound_circuits[index]
+                final_state = Statevector(bound_circuit_to_instruction(bound_circuit))
+                final_state.seed(self._rng)
+                if qargs:
+                    samples = final_state.sample_memory(shots=pub.shots, qargs=qargs)
```

Comment:
TODO For the future: we should add a `sample` method to Statevector (or make a helper function here) that directly samples in the BitArray format rather than constructing bit strings and converting them which is going to be more inefficient.